### PR TITLE
Comment Hiding + HTML Comments

### PR DESCRIPTION
**Pull request by Rahkeen**
This PR adds the ability to hide comment "subtrees".

Because of the way we parse, each comment is an indpendent entity with no parent-child relationship. However, we can use the comment order + indent level to determine if a comment is a child of some arbitrary parent comment. I setup a simple function to find all the children of a comment that would be hidden or displayed based on the selected parent. I also added a lil icon to show show if a comment is displayed or hidden.

Last thing I snuck in here was getting the comment HTML rather than the escaped string, so that we can parse comments as HTML (render links, formatting, etc).

### CODE REVIEW

**Comment by Rahkeen** [comment]
* **#109** <a href="https://app.graphite.dev/github/pr/EmergeTools/hackernews/109?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#108** <a href="https://app.graphite.dev/github/pr/EmergeTools/hackernews/108?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Rahkeen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: android/app/src/main/java/com/emergetools/hackernews/features/comments/CommentsScreen.kt [hunk 1]
```diff
@@ -235,21 +239,23 @@ private fun CommentsScreenLoadingPreview() {
...
+  when (state) {
+    is CommentState.Content -> {
+      val startPadding = (state.level * 16).dp
+      Column(
+        modifier = modifier
+          .padding(start = startPadding)
+          .fillMaxWidth()
+          .wrapContentHeight()
+          .clip(RoundedCornerShape(8.dp))
+          .clickable { onToggleHide(state) }
```
Comment (rbro112) [inline android/app/src/main/java/com/emergetools/hackernews/features/comments/CommentsScreen.kt:254]: Thoughts on clickable vs long click? No strong preference, but worried clicking might be too sensitive

**Review by rbro112 — APPROVED** [review]

File: android/app/src/main/java/com/emergetools/hackernews/features/comments/CommentsScreen.kt (hunk 1 above)
Comment (Rahkeen) [reply android/app/src/main/java/com/emergetools/hackernews/features/comments/CommentsScreen.kt:254]: Based on our convo, we'll keep this as is for now. Seems to be convention (according to the stock reddit app).

**Comment by Rahkeen** [comment]
### Merge activity

* **Aug 8, 6:45 PM EDT**: @Rahkeen started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/EmergeTools/hackernews/108).

**Pull request merged**